tests/ReturnTypeTest.php

```diff
@@ -1298,7 +1298,7 @@ function foo(): array {
                       }
                       return $arr;
                     }',
-                'error_message' => 'LessSpecificReturnStatement',
+                'error_message' => 'InvalidReturnStatement',
```

Comment:
This feels like a change that should be in latest Psalm 5.x — can you extract whatever changes you had to make in a separate PR please?

Reply:
I removed the BC break from this PR, this PR can now be directly merged in 5.x :)